data/scenarios/Challenges/_wave/wavebot.sw

```diff
@@ -0,0 +1,20 @@
+def doN = \n. \f. if (n > 0) {f; doN (n - 1) f} {}; end;
+
+def crossPath =
+    doN 6 move;
```

Comment:
I suspect this is needlessly slow because we do not inline it.
```suggestion
    // doN 6 move;
    move; move; move; move; move; move;
```

I also tried replacing `wait` with `turn forward`.

Subjectively, the simulation seemed to keep playing better without lag on my computer once inlined.

Reply:
Just following a [blog post about profiling with cabal](https://nikita-volkov.github.io/profiling-cabal-projects/), I can see some difference in the graphs:

| Non-Inlined | Inlined |
|--------|--------|
| ![swarm_noninlined](https://github.com/swarm-game/swarm/assets/44544735/359d4c80-ba14-4585-8efe-7a79c41293bf) | ![swarm_inlined](https://github.com/swarm-game/swarm/assets/44544735/7081f758-204f-4de3-9e52-ea120852612b) |

Reply:
@xsebek Great work looking at performance, memory profiling, etc.!  We haven't paid much attention to it so far but I'm sure there is tons of low-hanging fruit to improve performance.

I'm not sure inlining things in scenario programs is the right way to go, though.  Ideally we should be able to encourage nice, modular code and work hard to make that nice, abstracted, modular code run fast.

Reply:
@byorgey it’s definitely not ideal to do it manually, thats why I made an issue to do it automatically. 🙂

* #1557